app/helpers/projects_helper.rb

```diff
@@ -1,2 +1,134 @@
 module ProjectsHelper
+  # This method DOESN'T check that each row has required metadata fields such as collection_location, tissue_type.
+  # This can validate both samples that are already created, as well as samples that are new.
+  def validate_metadata_csv_for_project(project_samples, metadata, new_samples)
+    errors = []
+    warnings = []
+
+    if new_samples
+      # Require host_genome_name as a column.
+      unless metadata["headers"].include?("host_genome_name")
+        errors.push("host_genome_name column is required.")
+        return { "errors" => errors, "warnings" => warnings }
+      end
+    else
+      # Require sample_name as a column.
```

Comment:
So if new_samples is true, then sample_name is not required? How come?

Reply:
`host_genome_names` is required instead. For new_samples, assuming that the samples haven't been created yet. Later, will need both `project_id` and `host_genome_name` to figure out what metadata fields to use.